Widgets/WidgetVPNIntents.swift

```diff
@@ -22,99 +22,102 @@ import NetworkExtension
 import NetworkProtection
 import WidgetKit
 import Core
+import VPNWidgetSupport
 
 // MARK: - Enable & Disable
 
+/// App intent to disable the VPN
+///
+/// This is used in our Widget only.
+/// This is very similar to ``DisableVPNAppIntent``, but this can run in both widget and app,
+/// does not support continuation in the app and does not provide any result dialog.
+///
 @available(iOS 17.0, *)
-struct DisableVPNIntent: AppIntent {
+struct WidgetDisableVPNIntent: AppIntent {
 
-    static let title: LocalizedStringResource = "Disable VPN"
+    static let title: LocalizedStringResource = "Disable DuckDuckGo VPN"
     static let description: LocalizedStringResource = "Disables the DuckDuckGo VPN"
     static let openAppWhenRun: Bool = false
     static let isDiscoverable: Bool = false
+    static var authenticationPolicy: IntentAuthenticationPolicy = .requiresAuthentication
 
     @MainActor
     func perform() async throws -> some IntentResult {
         do {
-            DailyPixel.fire(pixel: .networkProtectionWidgetDisconnectAttempt)
+            DailyPixel.fireDailyAndCount(pixel: .networkProtectionWidgetDisconnectAttempt)
 
-            let managers = try await NETunnelProviderManager.loadAllFromPreferences()
-            guard let manager = managers.first else {
-                return .result()
-            }
-
-            manager.isOnDemandEnabled = false
-            try await manager.saveToPreferences()
-            manager.connection.stopVPNTunnel()
+            let controller = VPNWidgetTunnelController()
```

Comment:
I moved most of the shared logic into this controller and intents are now more specific to where they're triggered from, so we can fire specific pixels depending on source.